pkg/typesystem/typesystem_test.go

```diff
@@ -2819,8 +2840,12 @@ func TestIsPubliclyAssignable(t *testing.T) {
 			typesys := New(model)
 
 			ok, err := typesys.IsPubliclyAssignable(test.target, test.objectType)
-			require.NoError(t, err)
-			require.Equal(t, ok, test.result)
+			if test.expectedError != "" {
+				require.ErrorContains(t, err, test.expectedError)
```

Comment:
Since we don't check for error but instead check whether return is false in `hasPubliclyAssignedType, _ := typesys.IsPubliclyAssignable(relationReference, userType)`, should we add 
```suggestion
				require.False(ok)
				require.ErrorContains(t, err, test.expectedError)
```